integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java

```diff
@@ -515,76 +512,242 @@ public void testDomainK8SEventsDelete() {
+   * --set "domainNamespaceLabelSelector=weblogic-operator\=enabled"
    * }
    * </pre>
    * </p>
-   * Test verifies NamespaceWatchingStopped event is logged when operator stops watching a domain namespace.
    */
-  @Order(12)
-  @Test
-  @Disabled("Bug - OWLS-87181")
-  public void testK8SEventsStopWatchingNS() {
+  @Order(13)
+  @ParameterizedTest
+  @ValueSource(booleans = { true, false })
+  public void testK8SEventsStartStopWatchingNSWithLabelSelector(boolean enableClusterRoleBinding) {
+    logger.info("testing testK8SEventsStartStopWatchingNSWithLabelSelector with enableClusterRoleBinding={0}",
+        enableClusterRoleBinding);
     OffsetDateTime timestamp = now();
+
+    logger.info("Adding a new domain namespace in the operator watch list");
+    // label domainNamespace3
+    new Command()
+        .withParams(new CommandParams()
+            .command("kubectl label ns " + domainNamespace3 + " weblogic-operator=enabled --overwrite"))
+        .execute();
+
+    // Helm upgrade parameters
+    opParams = opParams
+        .domainNamespaceSelectionStrategy("LabelSelector")
+        .domainNamespaceLabelSelector("weblogic-operator=enabled")
+        .enableClusterRoleBinding(enableClusterRoleBinding);
+    upgradeAndVerifyOperator(opNamespace, opParams);
+
+    logger.info("verify NamespaceWatchingStarted event is logged in " + domainNamespace3);
+    checkEvent(opNamespace, domainNamespace3, null, NAMESPACE_WATCHING_STARTED, "Normal", timestamp);
+
+    // verify there is no event logged in domainNamespace4
+    logger.info("verify NamespaceWatchingStarted event is not logged in " + domainNamespace4);
+    assertFalse(domainEventExists(opNamespace, domainNamespace4, null, NAMESPACE_WATCHING_STARTED,
+        "Normal", timestamp), "domain event " + NAMESPACE_WATCHING_STARTED + " is logged in "
+        + domainNamespace4 + ", expected no such event will be logged");
+
+    timestamp = now();
+    logger.info("Removing domain namespace in the operator watch list");
+
+    // label domainNamespace3 to weblogic-operator=disabled
+    new Command()
+        .withParams(new CommandParams()
+            .command("kubectl label ns " + domainNamespace3 + " weblogic-operator=disabled --overwrite"))
+        .execute();
+
+    logger.info("verify NamespaceWatchingStopped event is logged");
+    checkNamespaceWatchingStoppedEvent(opNamespace, domainNamespace3, null, "Normal", timestamp,
+        enableClusterRoleBinding);
+  }
+
+  /**
+   * Test verifies the operator logs a NamespaceWatchingStarted event in the respective domain namespace
+   * when it starts watching a new domain namespace with domainNamespaceSelectionStrategy set to RegExp and
+   * operator logs a NamespaceWatchingStopped event in the respective domain namespace
+   * when it stops watching a domain namespace.
+   * The test upgrades the operator instance through helm to add or remove another domain namespace
+   * in the operator watch list.
+   *<p>
```

Comment:
can you please document how the operator uses RegExp(the same for Dedicated as well)  to start/stop watching namespace? A example would be nice.

Reply:
fixed